salesforce/generate-salesforce-refresh-token.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+function obtain_authorization_code {
+    instance_url=$1
+    client_key=$2
+    redirect_uri=$3
+
+    authorization_url="https://${instance_url}/services/oauth2/authorize?response_type=code&client_id=${client_key}&redirect_uri=${redirect_uri}"
+    printf "\nPlease log in to Salesforce and visit the following URL to obtain the authorization code:"
+    echo "$authorization_url"
+}
+
+function obtain_access_token {
+    instance_url=$1
+    client_key=$2
+    client_secret=$3
+    redirect_uri=$4
+    authorization_code=$5
+
+    token_url="https://${instance_url}/services/oauth2/token"
+    params=("code=${authorization_code}&grant_type=authorization_code&client_id=${client_key}&client_secret=${client_secret}&redirect_uri=${redirect_uri}")
+
+    response=$(curl -s -X POST "${token_url}" -d "${params[*]}")
+    if [ $? -ne 0 ]; then
+        echo "Failed to obtain refresh token. Response: $response" >&2
+        exit 1
+    fi
+
+    echo "$response"
```

Comment:
Why are we logging this response?

Reply:
The response contains the object which holds refreshToken, accessToken, expiry time and granted permission information.